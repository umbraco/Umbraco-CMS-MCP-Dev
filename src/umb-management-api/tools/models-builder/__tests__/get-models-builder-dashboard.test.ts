import GetModelsBuilderDashboardTool from "../get/get-models-builder-dashboard.js";
import { UmbracoManagementClient } from "@umb-management-client";
import { jest } from "@jest/globals";

describe("get-models-builder-dashboard", () => {
  let originalConsoleError: typeof console.error;
  let originalGetClient: typeof UmbracoManagementClient.getClient;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
    originalGetClient = UmbracoManagementClient.getClient;
  });

  afterEach(() => {
    console.error = originalConsoleError;
    UmbracoManagementClient.getClient = originalGetClient;
  });

  it("should get the models builder dashboard", async () => {
    const result = await GetModelsBuilderDashboardTool().handler(
      {},
      { signal: new AbortController().signal }
    );
    // Verify the handler response using snapshot
    expect(result).toMatchSnapshot();
  });

});