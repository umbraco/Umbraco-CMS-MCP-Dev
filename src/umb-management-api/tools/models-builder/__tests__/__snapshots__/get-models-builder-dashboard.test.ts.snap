// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-models-builder-dashboard should get the models builder dashboard 1`] = `
{
  "content": [
    {
      "text": "{"mode":"InMemoryAuto","canGenerate":false,"outOfDateModels":false,"lastError":null,"version":"16.1.1+7e82c25","modelsNamespace":"Umbraco.Cms.Web.Common.PublishedModels","trackingOutOfDateModels":false}",
      "type": "text",
    },
  ],
}
`;
